tests/test_riskpool_lifecycle.py

```diff
@@ -293,6 +293,367 @@ def test_suspend_resume(
+    print(bundle)
+
+    (
+        bundleId,
+        riskpoolId,
+        tokenId,
+        state,
+        filter,
+        capital,
+        lockedCapital,
+        balance,
+        createdAt,
+        updatedAt
+    ) = bundle
+
+    # check bundle values with expectation
+    assert bundleId == 1
+    assert riskpoolId == riskpool.getId()
+    assert owner != riskpool.owner()
+    print(riskpool.owner())
+
+    # verify that attributes directly from riskpool match with 
+    # bundle attributes from instance service
+    bundleFromCore = instanceService.getBundle(bundleId).dict()
+    print(bundleFromCore)
+
+    assert bundleFromCore["id"] == bundleId
+    assert bundleFromCore["riskpoolId"] == riskpoolId
+    assert bundleFromCore["tokenId"] == tokenId
+    assert bundleFromCore["state"] == state
+    assert bundleFromCore["filter"] == filter
+    assert bundleFromCore["capital"] == capital
+    assert bundleFromCore["lockedCapital"] == lockedCapital
+    assert bundleFromCore["balance"] == balance
+    assert bundleFromCore["createdAt"] == createdAt
+    assert bundleFromCore["updatedAt"] == updatedAt
+
+    componentOwnerService = instance.getComponentOwnerService()
+
+    # ensure that component owner may not suspend riskpool
+    assert owner != riskpoolKeeper
+    with brownie.reverts("ERROR:CMP-015:ACTIVE_INVALID_TRANSITION"):
+        instanceOperatorService.archive(riskpoolId, {'from':owner})
+
+    with brownie.reverts("ERROR:CMP-015:ACTIVE_INVALID_TRANSITION"):
+        componentOwnerService.archive(riskpoolId, {'from':riskpoolKeeper})
+
+    assert instanceService.getComponentState(riskpoolId) == 3
+
+    # ensure that instance operator may suspend riskpool
+    instanceOperatorService.suspend(riskpoolId, {'from':owner})
+    assert instanceService.getComponentState(riskpoolId) == 5
+
+    # ensure that instance operator may archive riskpool
+    instanceOperatorService.archive(riskpoolId, {'from':owner})
+    assert instanceService.getComponentState(riskpoolId) == 6
+
+    # ensure that riskpool actions are now blocked
+    with brownie.reverts("ERROR:RPS-002:RISKPOOL_NOT_ACTIVE"):
+        riskpool.createBundle(bytes(0), 50, {'from':bundleOwner})
+
+    with brownie.reverts("ERROR:RPS-002:RISKPOOL_NOT_ACTIVE"):
+        riskpool.fundBundle(bundleId, 10, {'from':bundleOwner})
+
+    with brownie.reverts("ERROR:RPS-002:RISKPOOL_NOT_ACTIVE"):
+        riskpool.defundBundle(bundleId, 10, {'from':bundleOwner})
+
+    with brownie.reverts("ERROR:RPS-002:RISKPOOL_NOT_ACTIVE"):
+        riskpool.lockBundle(bundleId, {'from':bundleOwner})
+
+    with brownie.reverts("ERROR:RPS-002:RISKPOOL_NOT_ACTIVE"):
+        riskpool.unlockBundle(bundleId, {'from':bundleOwner})
+
+    with brownie.reverts("ERROR:RPS-002:RISKPOOL_NOT_ACTIVE"):
+        riskpool.closeBundle(bundleId, {'from':bundleOwner})
+
+    with brownie.reverts("ERROR:RPS-002:RISKPOOL_NOT_ACTIVE"):
+        riskpool.burnBundle(bundleId, {'from':bundleOwner})
+

```

Comment:
suggestion: add check for transistion from archived to archived
expected revert message: invalid state transition, alternativel: source and target state identical